src/state/queries/usePostThread/queryCache.ts

```diff
@@ -85,10 +89,13 @@ export function createCacheMutator({
           /*
            * Update parent data
            */
-          parent.value.post = {
-            ...parent.value.post,
-            replyCount: (parent.value.post.replyCount || 0) + 1,
-          }
+          const shadow = dangerousGetPostShadow(parent.value.post)
+          const prevOptimisticCount = shadow?.optimisticReplyCount || 0
+          const prevReplyCount = parent.value.post.replyCount || 0
+          updatePostShadow(queryClient, parent.value.post.uri, {
+            // prefer optimistic count, if we already have some
+            optimisticReplyCount: (prevOptimisticCount || prevReplyCount) + 1,
+          })
```

Comment:
Would this make the priority a bit clearer 

```suggestion
          const prevOptimisticCount = shadow?.optimisticReplyCount
          const prevReplyCount = parent.value.post.replyCount
          updatePostShadow(queryClient, parent.value.post.uri, {
            // prefer optimistic count, if we already have some
            optimisticReplyCount: (prevOptimisticCount ?? prevReplyCount ?? 0) + 1,
          })
```